common/src/main/java/org/apache/rocketmq/common/BrokerConfig.java

```diff
@@ -47,6 +47,7 @@
     private boolean autoCreateSubscriptionGroup = true;
     private String messageStorePlugIn = "";
 
+    //thread counts for sending message, if large number threads are needed, we suggest setting useReentrantLockWhenPutMessage=true(in MessageStoreConfig)
```

Comment:
> OS is getting so smart that things may have changed considerably.
Exactly. That is why I wouldn't make such a suggestion in the code. You cannot bet it would perform better when you switch to lock mutex.